toolchain/check/keyword_modifier_set.h

```diff
@@ -70,10 +70,38 @@ class KeywordModifierSet {
   auto Remove(KeywordModifierSet set) -> void { set_ &= ~set.set_; }
 
   // Returns true if there's a non-empty set intersection.
-  constexpr auto HasAnyOf(KeywordModifierSet other) -> bool {
+  constexpr auto HasAnyOf(KeywordModifierSet other) const -> bool {
     return set_ & other.set_;
   }
 
+  // Return a builder that implicitly converts to the specified enumeration type
+  // mapping the cases passed to the `Case`s specified. For example:
+  //   ```
+  //   SomeEnum e = set.ToEnumerator(SomeEnum::Default)
+  //                    .Case(KeywordModifierSet::A, SomeEnum::A)
+  //                    .Case(KeywordModifierSet::B, SomeEnum::B);
+  //   ```
+  template <typename T>
+  auto ToEnumerator(T default_value) {
```

Comment:
```suggestion
  auto ToEnumerator(T default_value) -> auto {
```

I checked with zygoloid and he's okay with the nested scope, but suggested `-> auto` here to make it clear this is a deliberate return and not forgotten.